salt/states/dockerng.py

```diff
@@ -202,6 +202,15 @@ def _image_get(path, default=None):
                     env_ptr = env_diff.setdefault(item, {})
                     env_ptr.setdefault('old', {})[key] = actual_val
                     env_ptr.setdefault('new', {})[key] = data[key]
+            # Comparing list vice-versa
+            for key in actual_env:
+                # Need to skip PATH and no_proxy variables.
+                if key not in ['PATH', 'no_proxy']:
```

Comment:
These 2 variables are set by docker engine by default. 'docker inspect' shows these variables in ENV section as below. So these variables are included in list of actual_env variables and if we don't skip them container will get restarted every time even if there is no change in user defined environment variables.

"Env": [
                "VAR3=value33",
                "no_proxy=*.local, 169.254/16",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],